themes/src/main/resources/theme/base/admin/resources/js/controllers/clients.js

```diff
@@ -1096,6 +1096,8 @@ module.controller('ClientDetailCtrl', function($scope, realm, client, templates,
             }
             $scope.client.publicClient = false;
             $scope.client.serviceAccountsEnabled = true;
+        } else if ($scope.client.bearerOnly) {
```

Comment:
Yep, you're right. I missed the if statement that makes sure this is not used for clients with authz services enabled.